src/celengine/astrocat.h

```diff
@@ -0,0 +1,69 @@
+
+#pragma once
+
+#include <climits>
+#include <string>
+#include <celutil/util.h>
+#include <fmt/printf.h>
+
+class AstroCatalog
+{
+ public:
+    typedef unsigned int IndexNumber;
+    static const IndexNumber InvalidIndex = UINT_MAX;
+    virtual IndexNumber getIndexNumberByName(const std::string&) = 0;
+    virtual std::string getNameByIndexNumber(IndexNumber) = 0;
+    virtual const std::string& getName() const = 0;
+};
+
+class SimpleAstroCatalog : public AstroCatalog
+{
+ protected:
+    std::string m_prefix;
+ public:
+    SimpleAstroCatalog(const std::string& pref) { m_prefix = pref; }
+    virtual IndexNumber getIndexNumberByName(const std::string&);
```

Comment:
please prefer C++11 syntax for overridden methods:
```
IndexNumber getIndexNumberByName(const std::string&) overrride;
```

With it you don't introduce new virtual functions by mistake.